crates/energy/src/battery.rs

```diff
@@ -0,0 +1,170 @@
+use bevy::prelude::*;
+
+pub(crate) struct BatteryPlugin;
+
+impl Plugin for BatteryPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(discharge_battery);
+    }
+}
+
+/// The rate at which the battery discharges in Joules per second.
+const DISCHARGE_RATE: f64 = 30_000.;
+/// The minimum energy level in Joules at which the unit can still move.
+const MIN_MOVE_ENERGY: f64 = 0.;
+/// The minimum energy level in Joules at which the unit can still attack.
+const MIN_ATTACK_ENERGY: f64 = 20_000.;
+/// The minimum energy level in Joules at which factories can still produce units.
+const MIN_FACTORY_ENERGY: f64 = 10_000_000.;
+/// The default capacity of the battery in Joules.
+const DEFAULT_CAPACITY: f64 = 100_000_000.; // 100 Mj
+
+/// The battery component is used to store the energy level of an entity.
+#[derive(Component, Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Battery {
+    /// The maximum capacity of the battery in joules.
+    capacity: f64,
+
+    /// The current energy level of the battery in joules.
+    energy: f64,
+}
+
+impl Default for Battery {
+    fn default() -> Self {
+        Self {
+            capacity: DEFAULT_CAPACITY,
+            energy: DEFAULT_CAPACITY,
+        }
+    }
+}
+
+impl Battery {
+    pub fn new(capacity: f64, energy: f64) -> Self {
```

Comment:
In this PR, I would remove `pub` from methods, structs, etc. where it is not necessary for compilation to succeed. Also, I would remove all unused methods and constants (to avoid merging dead code). We can implement action blocking (e.g. movement) in a follow up PR.

Reply:
If you want me to do it in another pr I could. Was gonna do this one. Would that be preferable?

Reply:
Gonna do it in another one.